contracts/account/extensions/AccountERC6900.sol

```diff
@@ -0,0 +1,494 @@
+            require(
+                !validation.selectors.contains(executionSelector),
+                ERC6900Utils.ERC6900MissingValidationForSelector(executionSelector)
+            );
+        }
+        // If the selector being checked is execute or executeBatch,
+        // it MUST perform additional checking on target.
+        if (executionSelector == IERC6900ModularAccount.execute.selector) {
+            (address target, , ) = abi.decode(userOp.callData[4:], (address, uint256, bytes));
+            require(target != address(this), ERC6900Utils.ERC6900InvalidExecuteTarget());
+        }
+        if (executionSelector == IERC6900ModularAccount.executeBatch.selector) {
+            Call[] memory calls = abi.decode(userOp.callData[4:], (Call[]));
+            for (uint256 i = 0; i < calls.length; i++) {
+                require(calls[i].target != address(this), ERC6900Utils.ERC6900InvalidExecuteTarget());
+            }
+        }
+        return
+            IERC6900ValidationModule(module).validateUserOp(entityId, userOp, _signableUserOpHash(userOp, userOpHash));
+    }
+
+    /**
+     * @dev Lowest-level signature validation function. See {ERC7739-_rawSignatureValidation}.
+     *
+     * This function delegates the signature validation to a validation module if the first 20 bytes of the
+     * signature correspond to an installed validator module.
+     *
+     * See {_extractSignatureValidator} for the module extraction logic.
+     */
+    function _rawSignatureValidation(
+        bytes32 hash,
+        bytes calldata signature
+    ) internal view virtual override returns (bool) {
+        if (signature.length < 25) return false;
+        (ModuleEntity validationModuleEntity, bytes calldata innerSignature) = _extractSignatureValidator(signature);
+        (address module, uint32 entityId) = validationModuleEntity.unpack();
+        // If the validation function is attempted to be used for signature validation
+        // and the flag isSignatureValidation is set to false, validation MUST revert
+        require(
+            !_validations[validationModuleEntity].validationFlags.isSignatureValidation(),
+            ERC6900Utils.ERC6900BadSignatureValidation(validationModuleEntity)
+        );
+
+        return
+            IERC6900ValidationModule(module).validateSignature(
+                address(this),
+                entityId,
+                msg.sender,
+                hash,
+                innerSignature
+            ) == IERC1271.isValidSignature.selector;
+    }
+
+    /**
+     * @dev ERC-6900 execution logic.
+     *
+     * Reverts if the call type is not supported.
+     */
+    function _execute(address target, uint256 value, bytes calldata data) internal virtual returns (bytes memory) {
+        (bool success, bytes memory returndata) = target.call{value: value}(data);
+
+        require(success, ERC6900Utils.ERC6900ExecutionFailed(target, value, data));
+        return returndata;
```

Comment:
Is the `ERC6900Utils.ERC6900ExecutionFailed` standard ? Otherwise I think we should just do something that bubble the revert reason, for example using `Address.verifyCallResult`

@arr00 do you know what the ERC says ?

Reply:
There are no errors specified in the ERC.

Reply:
so we should probably be doing
```suggestion
        (bool success, bytes memory returndata) = target.call{value: value}(data);
        return Address.verifyCallResult(success, returndata);
```

Reply:
Thank you, updated in [Bubble revert reason when execute fails](https://github.com/OpenZeppelin/openzeppelin-community-contracts/pull/80/commits/ace9b0300545008a3d312bf6cee0cba999d88d13).